### DISCUSSION TITLE
route guard bug

### DISCUSSION THREAD

**Discussion by etherbob** [vuejs/router#2237] [Help and Questions]
I had a onBeforeUnmount callback on a component with an error. This raised an error when navigating, however instead of showing me the error Vue-router rescued and told me a required param was missing from the route I was navigating from. I'm wondering if this is expected behavior, a bug, or a misconfiguration on my part and if anyone here has any suggestions/insights to help me narrow that down before I open an issue.

**Comment by posva** [comment]
component mounting and navigation are usually disconnected. A code sample would help. You can build it with [playground](https://play.vuejs.org/#eNq1U8lu2zAQ/RVCLSAHsMVmQQ6CkqZNgy7oEqRFL1UPjEXbTCSS4OK4MPTvHZEUZceJgR56MExxZt68efO4ThrCeHankzxhjRTKoDWaKkoMfSMlatFMiQalS0vTkocEJayhKkQy7D+HcFfXx4iUma8teUQdwe+g5AhlVtORL/ffjbDcjNIXUJYeJOMklAO3wtBG1lB/3iUW0v3BQRsl+Pz80ipFeaCGJDGLvMAhhtZr9NIFspmt62sIorZ1MNjjFLbu8WoWTnC+ccw+M36PciPOymSNOGlojtKK6MWtIKpKUVsm5+8FMgLheFvgobQHxhH5n3uMYSBFGp3DaliVo8OO/m5XfPif+x490/doX98CB3GLzmh91Of/ZPQB4ZAWwgWOmwYDeHM8ac8vtBHqzwemDfyNw6XH3bTtJNozGvRdz9z17626dTuYVnAdbKXRGfrVcY1SLURDnUpgN5RiOE4FtODgxPxRl3a8Vbktsi8fBM1Z9XovWMl/PyKngNymBKN112/h1cmfkmx04Cj50UrewgssOV05hSo6I7buoWEPO+LsPsmKLcN2o2viu/M38dX51O1N+91MGiJh24IDvpugDAFdJgDo8ctk2Gt3XSYLY6TOMbZc3s8z0A0PGRcn2XH2Cnpqs3GbUd1MbpV40M5eZeLE6LAvIAlXdGmEqPWESPZci53Ei9PsNDsE499iQMeMV3TlsDtoGL2FMY2Gpc3Y/NGQ3apZTdU3aRgsdWtYUtfi4ZO7M8rSSHS6oNP7J+7v9MpTvlYUplvSjeEMUXNqfPjq+1e6gnMMNqKyNWTvCd5QLWrbcfRpby2vgPZGnmP70e2M8fkPfbUylOt+qI6oU8Plu0Ve7hl9oHucnUQV27/hwj7r)

**Reply by etherbob** [reply]
I wasn't able to replicate it in a simpler app. I'll try to figure out what's special about the spot in our app I saw that behavior.
